### DISCUSSION TITLE
Migration Doc Considerations

### DISCUSSION THREAD

**Discussion by zsamji** [bcgov/sso-keycloak#140] [Ideas]
Before you migrate you should know about

1. clients
2. roles
3. groups
4. users
5. mappers
6. idps

@arcshiftsolutions  and @junminahn  please add any documentaiton you want to review here before we publish to wiki

**Comment by zsamji** [comment]
how to describe/share infor on " override sub in the payload token to preferred_username so we could prevent them from using keycloak ids in the first place"

**Reply by zsamji** [reply]
@arcshiftsolutions with respect to a comment in the SSO channel, this discussion thread or a stack overflow one could be a good place to share the updates on the ```sub``` pattern in gold. Thoughts?

**Comment by arcshiftsolutions** [comment]
@zsamji Additional:

- Custom authentication flows
- Realm level setting (token timeouts/login with email/duplicate email etc.)
- Scopes 
- User Federation

**Comment by arcshiftsolutions** [comment]
I'm wondering where we're going to document our tooling we're going to provide? For example the scripts we have to migrate users from a KC realm to another.

**Comment by zsamji** [comment]
Closing the loop; we've added details in stackover flow and set up direct RC conversations.

https://stackoverflow.developer.gov.bc.ca/questions/864

https://stackoverflow.developer.gov.bc.ca/questions/915/
